plugin crashes when rate_limiter with action to "log_only" is set

### Terraform Version

```shell
v1.0.11
```

### fastly provider version

```shell
4.3.3
```

### Affected Fastly Terraform Resource(s)

- `fastly_service_vcl`
  -  `rate_limiter`

When creating a `rate_limiter` with option of `log_only` for `action`, the plugin crashed every time trying to run apply even tho it was fine running the plan.


### Terraform Configuration Files

```hcl
       rate_limiter {
           action               = "log_only"
           client_key           = "client.ip,req.http.User-Agent"
           http_methods         = "GET,PUT,TRACE,POST,HEAD,DELETE,PATCH,OPTIONS"
           logger_type          = "datadog"
           name                 = "rate limit"
           penalty_box_duration = 2
           ratelimiter_id       = (known after apply)
           rps_limit            = 1000
           window_size          = 60
       }

```

### Expected Behavior

 Terraform applied successfully

### Actual Behavior

Plugin crashed hard

### Debug Output (optional)

```
module.test_service.fastly_service_vcl.service: Modifying... [id=3B......]
╷
│ Error: Plugin did not respond
│
│   with module.test_service.fastly_service_vcl.service,
│   on ../shared/modules/fastly_service/main.tf line 165, in resource "fastly_service_vcl" "service":
│  165: resource "fastly_service_vcl" "service" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-fastly_v4.3.3 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 86 [running]:
github.com/fastly/terraform-provider-fastly/fastly.(*RateLimiterAttributeHandler).buildUpdateERLInput(0x0?, {0xc000294678?, 0x0?}, {0x0?, 0x0?}, 0x0?, 0x0?, 0x0?)
	github.com/fastly/terraform-provider-fastly/fastly/block_fastly_service_ratelimiter.go:394 +0x90c
github.com/fastly/terraform-provider-fastly/fastly.(*RateLimiterAttributeHandler).Update(0x19a6f40?, {0xc000153d60?, 0x0?}, 0xc000598c00, 0xc000150100?, 0xc0003efcc0?, 0x54, 0x19e7e26?)
	github.com/fastly/terraform-provider-fastly/fastly/block_fastly_service_ratelimiter.go:243 +0x36f
github.com/fastly/terraform-provider-fastly/fastly.(*blockSetAttributeHandler).Process(0xc000031b90, {0x1b34588, 0xc00069c060}, 0x151?, 0xc0008025f0?, 0x225b108?)
	github.com/fastly/terraform-provider-fastly/fastly/service_crud_attribute_definition.go:121 +0x3dd
github.com/fastly/terraform-provider-fastly/fastly.resourceServiceUpdate({0x1b34588, 0xc00069c060}, 0xc000598c00, {0x1874aa0?, 0xc000290188}, {0x1b31350, 0x205a9a0})
	github.com/fastly/terraform-provider-fastly/fastly/base_fastly_service.go:400 +0x1a07
github.com/fastly/terraform-provider-fastly/fastly.resourceUpdate.func1({0x1b34588?, 0xc00069c060?}, 0x0?, {0x1874aa0?, 0xc000290188?})
	github.com/fastly/terraform-provider-fastly/fastly/base_fastly_service.go:216 +0x35
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc0000a5b20, {0x1b345c0, 0xc000284510}, 0xd?, {0x1874aa0, 0xc000290188})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:741 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0000a5b20, {0x1b345c0, 0xc000284510}, 0xc000401380, 0xc0000c0480, {0x1874aa0, 0xc000290188})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/resource.go:847 +0x82c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000d9e0, {0x1b345c0?, 0xc0001ebfb0?}, 0xc0002c4410)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.26.1/helper/schema/grpc_provider.go:1021 +0xe3c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc0000a3680, {0x1b345c0?, 0xc0001ea120?}, 0xc000592230)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:818 +0x574
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x19a6de0?, 0xc0000a3680}, {0x1b345c0, 0xc0001ea120}, 0xc000592000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000001e0, {0x1b380c0, 0xc0002a09c0}, 0xc0007ca000, 0xc000412720, 0x2062980, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc0000001e0, {0x1b380c0, 0xc0002a09c0}, 0xc0007ca000, 0x0)
	google.golang.org/grpc@v1.51.0/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-fastly_v4.3.3 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

```

### Steps to Reproduce


1. `terraform apply`

--- a/fastly/block_fastly_service_ratelimiter.go
+++ b/fastly/block_fastly_service_ratelimiter.go
@@ -42,7 +42,7 @@ func (h *RateLimiterAttributeHandler) GetSchema() *schema.Schema {
 			Type:        schema.TypeString,
 			Required:    true,
 			Description: "The action to take when a rate limiter violation is detected (one of: log_only, log_only, response_object)",
-			ValidateDiagFunc: func(i interface{}, _ cty.Path) diag.Diagnostics {
+			ValidateDiagFunc: func(i any, _ cty.Path) diag.Diagnostics {
 				for _, a := range gofastly.ERLActions {
 					if i.(string) == string(a) {
 						return nil
@@ -55,7 +55,7 @@ func (h *RateLimiterAttributeHandler) GetSchema() *schema.Schema {
 			Type:        schema.TypeString,
 			Required:    true,
 			Description: "Comma-separated list of VCL variables used to generate a counter key to identify a client",
-			ValidateDiagFunc: func(i interface{}, _ cty.Path) diag.Diagnostics {
+			ValidateDiagFunc: func(i any, _ cty.Path) diag.Diagnostics {
 				if strings.Contains(i.(string), " ") {
 					return diag.Errorf("invalid client_key: should contain no spaces")
 				}
@@ -72,7 +72,7 @@ func (h *RateLimiterAttributeHandler) GetSchema() *schema.Schema {
 			Type:        schema.TypeString,
 			Required:    true,
 			Description: "Comma-separated list of HTTP methods to apply rate limiting to",
-			ValidateDiagFunc: func(i interface{}, _ cty.Path) diag.Diagnostics {
+			ValidateDiagFunc: func(i any, _ cty.Path) diag.Diagnostics {
 				v := i.(string)
 				if strings.Contains(v, " ") {
 					return diag.Errorf("invalid http_methods: should contain no spaces")
@@ -87,7 +87,7 @@ func (h *RateLimiterAttributeHandler) GetSchema() *schema.Schema {
 			Type:        schema.TypeString,
 			Optional:    true,
 			Description: "Name of the type of logging endpoint to be used when action is log_only (one of: azureblob, bigquery, cloudfiles, datadog, digitalocean, elasticsearch, ftp, gcs, googleanalytics, heroku, honeycomb, http, https, kafka, kinesis, logentries, loggly, logshuttle, newrelic, openstack, papertrail, pubsub, s3, scalyr, sftp, splunk, stackdriver, sumologic, syslog)",
-			ValidateDiagFunc: func(i interface{}, _ cty.Path) diag.Diagnostics {
+			ValidateDiagFunc: func(i any, _ cty.Path) diag.Diagnostics {
 				for _, l := range gofastly.ERLLoggers {
 					if i.(string) == string(l) {
 						return nil
@@ -156,7 +156,7 @@ func (h *RateLimiterAttributeHandler) GetSchema() *schema.Schema {
 			Type:        schema.TypeInt,
 			Required:    true,
 			Description: "Number of seconds during which the RPS limit must be exceeded in order to trigger a violation (one of: 1, 10, 60)",
-			ValidateDiagFunc: func(i interface{}, _ cty.Path) diag.Diagnostics {
+			ValidateDiagFunc: func(i any, _ cty.Path) diag.Diagnostics {
 				for _, w := range gofastly.ERLWindowSizes {
 					if i.(int) == int(w) {
 						return nil
@@ -347,7 +347,7 @@ func (h *RateLimiterAttributeHandler) buildUpdateERLInput(rateLimiterID, service
 		ERLID: rateLimiterID,
 	}
 
-	// NOTE: When converting from an interface{} we lose the underlying type.
+	// NOTE: When converting from an `any` type, we lose the underlying type.
 	// Converting to the wrong type will result in a runtime panic.
 
 	if v, ok := modified["action"]; ok {

--- a/fastly/block_fastly_service_ratelimiter.go
+++ b/fastly/block_fastly_service_ratelimiter.go
@@ -391,11 +391,14 @@ func (h *RateLimiterAttributeHandler) buildUpdateERLInput(rateLimiterID, service
 	}
 
 	if v, ok := modified["response"]; ok {
-		m := v.([]any)[0].(map[string]any)
-		input.Response = &gofastly.ERLResponseType{
-			ERLContent:     m["content"].(string),
-			ERLContentType: m["content_type"].(string),
-			ERLStatus:      m["status"].(int),
+		s := v.([]any)
+		if len(s) > 0 {
+			m := s[0].(map[string]any)
+			input.Response = &gofastly.ERLResponseType{
+				ERLContent:     m["content"].(string),
+				ERLContentType: m["content_type"].(string),
+				ERLStatus:      m["status"].(int),
+			}
 		}
 	}
 

--- a/fastly/block_fastly_service_ratelimiter.go
+++ b/fastly/block_fastly_service_ratelimiter.go
@@ -437,16 +437,19 @@ func flattenRateLimiter(remoteState []*gofastly.ERL, _ ServiceMetadata) []map[st
 			"name":                 o.Name,
 			"penalty_box_duration": o.PenaltyBoxDuration,
 			"ratelimiter_id":       o.ID,
-			"response": []map[string]any{
+			"response_object_name": o.ResponseObjectName,
+			"rps_limit":            o.RpsLimit,
+			"window_size":          int(o.WindowSize),
+		}
+
+		if o.Response != nil {
+			data["response"] = []map[string]any{
 				{
 					"content":      o.Response.ERLContent,
 					"content_type": o.Response.ERLContentType,
 					"status":       o.Response.ERLStatus,
 				},
-			},
-			"response_object_name": o.ResponseObjectName,
-			"rps_limit":            o.RpsLimit,
-			"window_size":          int(o.WindowSize),
+			}
 		}
 
 		result = append(result, data)

